.gitignore

```diff
@@ -334,78 +334,6 @@ opal/tools/wrappers/opal.pc
 opal/util/show_help_lex.c
 opal/util/keyval/keyval_lex.c
 
-test/monitoring/aggregate_profile.pl
```

Comment:
I noticed that the ignore file was out of date when I ran `make check` locally. It missed a few test artifacts, and also contained duplicates. Since I was here, I reorganized the file list.

Reply:
I'm not arguing against updates to .gitignore. But please detach them from this PR.